### PR TITLE
Script code check improvements

### PR DESCRIPTION
**Pull request by filipw**
- instead of excluding some diagnostics in the `CodeCheckService` (which was a hack), moved the suppression to the compilation options of the `ScriptProjectSystem`
 - we do not need to suppress `CS8099` anymore, because we are setting `SourceFileResolver` on the compilation (beforehand, it was missing, which generated the diagnostic).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,6 +31,7 @@ buildlog
 /omnisharp*.tar.gz
 scripts/Omnisharp*
 .msbuild-*/
+packages/
```
Comment (DustinCampbell) [inline .gitignore]: Was this intentional?

Comment (filipw) [inline .gitignore]: ugh, I somehow had that folder in the root. after git clean and rebuild, it's gone, must have been a leftover from a long ago; I will revert this

Comment (filipw) [inline .gitignore]: removed

**Comment by david-driscoll** [comment]
Mac build fell over, restarted.

**Comment by DustinCampbell** [comment]
Yeah, I restarted the Mac build as well this morning. I guess Travis is having trouble?

**Comment by DustinCampbell** [comment]
OK. I'm going to assume this is good. It's passed lots of times at this point. However, it looks like Travis is dropping PR events and not triggering.

**Comment by filipw** [comment]
thanks

**Pull request merged**